### ISSUE TITLE
Adding origins dynamically

### ISSUE THREAD

**Issue by scottbarrow** [cyu/rack-cors#50]
Does anyone know if there's there a way to dynamically add origins to the allowed list without having to restart the server?

Thanks

**Comment by faragorn** [comment]
I am not sure if this issue is still relevant, but here is the solution that worked for me:
```ruby
allow do
        origins do |source, env|
          # this proc should return true or false
          # You can dynamically check the database/redis or any other storage for your origins
          Database.where('redirect_uri  ~* ?', "^#{source}").exists?
        end
        resource '/api/v0/*', headers: :any, methods: [:get, :delete, :put, :post, :options]
        resource '/api/oauth/*', headers: :any, methods: [:get, :delete, :post, :options]
      end
```

**Comment by medBouzid** [comment]
@faragorn is your solution supposed to check for the customers who are allowed to use the api ? I have a web widget where my customers are allowed to send post requests to my server and I would like to prevent any "not allowed" request from getting a response

**Comment by faragorn** [comment]
@medBouzid Yes, in my solution it is done dynamically, and sources are checked in the database. But generally that's what `origins` are for, if request origin is not listed in the `origins` it will not be allowed.

**Comment by MaxSandulsky** [comment]
Hi every one, sorry for the stupid question, is there a way to update list of allowed origins on fly when server is already running? I need this to be able to integrate client custom APIs.

**Comment by jub0bs** [comment]
FWIW, here is a post about the approach I followed to make CORS middleware reconfigurable in my Go library: https://jub0bs.com/posts/2024-05-14-reconfigurable-cors-middleware/. Perhaps it can serve as inspiration for rack-cors.
